ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_client.py

```diff
@@ -130,8 +131,17 @@ def get_workflow_state(
             exist.
 
         """
-        state = self.__obj.get_orchestration_state(instance_id, fetch_payloads=fetch_payloads)
-        return WorkflowState(state) if state else None
+        try:
+            state = self.__obj.get_orchestration_state(instance_id, fetch_payloads=fetch_payloads)
+            return WorkflowState(state) if state else None
+        except RpcError as error:
+            if 'no such instance exists' in error.details():
+                self._logger.warning(f'Workflow instance not found: {instance_id}')
+                return None
```

Comment:
Does the RpcError raised by Dapr include the `NOT_FOUND` status code? Looks like you can do a `e.code()`  when looking at a real error? I'm not very familiar with `grpc` code.

I'm wondering if it's better to check by status code rather than string matching on the message?

Reply:
Unfortunately, this error hasn't been converted yet to the Rich Error model and it returns the default status `UNKNOWN`, so we have to check for the status message, for now.